test/passing/types.ml

```diff
@@ -50,3 +50,31 @@ val x :
     (* boooooom *)
     (** foooooooooooooooo foooooooooooooooooooooooo fooooooooooooooooooooooo
         fooooooooooooooooooo fooooooooooooooo*) ]
+
+type voting_period =
+      Tezos_client_alpha.Proto_alpha.Alpha_context.Voting_period.kind =
+  | Proposal
+  | Testing_vote
+
+(** foooooooo *)
+type voting_period =
+      (* foooooooooooo *)
+      (* foooooooooo ooooooooooooooooo ooooooooooooo *)
+      Tezos_client_alpha.Proto_alpha.Alpha_context.Voting_period.kind
+      (* fooooooooooooooo oooooooooooooooooooo ooooooooooooooooooooo *)
+      (* fooooo *) =
+  (* foooooooooo *)
+  | Proposal
+  | Testing_vote  (** fooooooooooo *)
+
+type ( 'context
+     , 'f_in
+     , 'f_out
+     , 'captured_types
+     , 'markers_in
+     , 'markers_out
+     , 'list_constraint ) templ_matcher =
+  { on_objc_cpp: 'context -> 'f_in
+  ; on_objc_cpp: 'context -> 'f_in
+  ; on_objc_cpp: 'context -> 'f_in
+  ; on_objc_cpp: 'context -> 'f_in }
```

Comment:
Could you add this case with type extension because it does not share ast and formatting logic:

```
type ( 'context
     , 'f_in
     , 'f_out
     , 'captured_types
     , 'markers_in
     , 'markers_out
     , 'list_constraint ) templ_matcher +=
  | On_objc_cpp : 'context -> 'f_in
  | On_objc_cpp : 'context -> 'f_in
  | On_objc_cpp : 'context -> 'f_in
  | On_objc_cpp : 'context -> 'f_in
```

Reply:
done